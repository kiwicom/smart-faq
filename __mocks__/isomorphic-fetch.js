// @flow

import { loginEndpoint, socialLoginEndpoint } from '../src/helpers/Requests';
import {
  goodEmail,
  goodPassword,
  defaultToken,
  defaultUserId,
  goodPayload,
} from './constants';

type Response = {
  json: () => Promise<{}>,
  status: string | number,
  statusText: string,
  headers: {},
  config: {},
};
type RequestType = {
  body: string,
};

const makeResponse = (
  status = 200,
  statusText = 'OK',
  data = {},
): Response => ({
  json: () => new Promise(resolve => resolve(data)),
  status,
  statusText,
  headers: {},
  config: {},
});

export default function fetch(
  url: string,
  obj: RequestType,
): Promise<Response | {}> {
  if (url === loginEndpoint) {
    const { login, password } = JSON.parse(obj.body);
    if (login === goodEmail && password === goodPassword) {
      return Promise.resolve(
        makeResponse(200, 'OK', {
          token: defaultToken,
          user_id: defaultUserId,
        }),
      );
    } else {
      return Promise.resolve(
        makeResponse(200, 'OK', {
          error_code: 'INVALID_ARGUMENT_LOGIN',
          message: 'Invalid email address provided.',
        }),
      );
    }
  } else if (url === socialLoginEndpoint) {
    const payload = JSON.parse(obj.body);
    if (JSON.stringify(payload) === JSON.stringify(goodPayload)) {
      return Promise.resolve(
        makeResponse(200, 'OK', {
          authorization_url: 'https://www.kiwi.com/en/?user_id=AAABBBCCC',
        }),
      );
    } else {
      return Promise.resolve(
        makeResponse(200, 'OK', {
          error_code: 'INVALID_ARGUMENT_IN_PAYLOAD',
          message: 'Invalid payload argument.',
        }),
      );
    }
  }
  return Promise.resolve({});
}
