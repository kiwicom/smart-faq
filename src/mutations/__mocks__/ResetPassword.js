// @flow

export default () => Promise.resolve();
