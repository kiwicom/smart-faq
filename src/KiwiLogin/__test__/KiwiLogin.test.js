// @flow
import * as React from 'react';
import { shallow } from 'enzyme';

import { RawKiwiLogin } from '../index';
import { defaultToken, defaultUserId } from '../../../__mocks__/constants';

describe('KiwiLogin', () => {
  const props = {
    onLogin: async () => ({
      user_id: defaultUserId,
      token: defaultToken,
    }),
    history: {
      push: jest.fn(),
    },
  };
  const e = {
    preventDefault: jest.fn(),
  };

  const wrapper = shallow(<RawKiwiLogin {...props} />);
  it('should change state', () => {
    const Login = wrapper.instance();
    const state = Login.state;
    expect(state).toEqual({ email: '', password: '', showError: false });

    wrapper.find(`[type="email"]`).simulate('change', {
      target: { name: 'email', value: 'email@kiwi.com' },
    });
    wrapper.find(`[type="password"]`).simulate('change', {
      target: { name: 'password', value: 'myRealPassword' },
    });

    expect(Login.state).toMatchSnapshot();
  });

  it('should handle sign in', async () => {
    const Login = wrapper.instance();
    await Login.handleSignIn(e);
    expect(e.preventDefault.mock.calls).toHaveLength(1);
    expect(Login.props.history.push.mock.calls[0][0]).toBe('/faq/');
  });

  it('should handle error in sign in', async () => {
    const errorProps = {
      history: {
        push: jest.fn(),
      },
      onLogin: () => new Promise((resolve, reject) => reject()),
    };

    const LoginWrapper = shallow(<RawKiwiLogin {...errorProps} />);
    const Login = LoginWrapper.instance();
    expect(Login.state.showError).toBe(false);

    await Login.handleSignIn(e);

    expect(Login.props.history.push.mock.calls).toHaveLength(0);
    expect(Login.state.showError).toBe(true);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
