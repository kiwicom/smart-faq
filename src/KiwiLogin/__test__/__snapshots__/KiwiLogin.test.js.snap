// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KiwiLogin should change state 1`] = `
Object {
  "email": "email@kiwi.com",
  "isLoading": false,
  "password": "myRealPassword",
  "showError": false,
}
`;

exports[`KiwiLogin should match snapshot 1`] = `
<div
  className="jsx-30246891 KiwiLogin"
>
  <CloseButton />
  <withRouter(BackButton)
    prevScreen="/sign-in"
  />
  <div
    className="jsx-30246891 picture"
  >
    <Illustration
      name="Help"
      size="medium"
    />
  </div>
  <div
    className="jsx-30246891 main"
  >
    <p
      className="jsx-30246891 title"
    >
      <Text
        html={false}
        t="smartfaq.kiwi_login.title"
        values={Object {}}
      />
    </p>
    <MediaQuery
      query="(min-width: 480px) and (min-height: 480px)"
      values={Object {}}
    >
      <Text
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.kiwi_login.subtitle"
          values={Object {}}
        />
      </Text>
    </MediaQuery>
    <MediaQuery
      query="only screen and (max-width: 480px) and (orientation: portrait)"
      values={Object {}}
    >
      <Text
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.kiwi_login.subtitle"
          values={Object {}}
        />
      </Text>
    </MediaQuery>
    <form
      className="jsx-30246891"
      onSubmit={[Function]}
    >
      <label
        className="jsx-30246891"
        htmlFor="email"
      >
        <Text
          html={false}
          t="smartfaq.kiwi_login.email_label"
          values={Object {}}
        />
        <div
          className="jsx-30246891 input"
        >
          <InputText
            dataCy="input-email"
            name="email"
            onChange={[Function]}
            placeholder="e.g. your@email.com"
            type="email"
            value="email@kiwi.com"
          />
        </div>
      </label>
      <label
        className="jsx-30246891"
        htmlFor="password"
      >
        <Text
          html={false}
          t="smartfaq.kiwi_login.password_label"
          values={Object {}}
        />
        <div
          className="jsx-30246891 input"
        >
          <InputText
            dataCy="input-password"
            name="password"
            onChange={[Function]}
            type="password"
            value="myRealPassword"
          />
        </div>
      </label>
      <div
        className="jsx-30246891 forgotPassword"
        data-cy="link-forgotten-password"
      >
        <TextLink
          external={false}
          onClick={[Function]}
          type="primary"
        >
          <Text
            html={false}
            t="smartfaq.kiwi_login.link_to_forgotten_password"
            values={Object {}}
          />
        </TextLink>
      </div>
      <span
        className="jsx-30246891 signIn"
        data-cy="btn-sign-in"
      >
        <Button
          block={true}
          onClick={[Function]}
        >
          <Text
            html={false}
            t="smartfaq.kiwi_login.sign_in_button"
            values={Object {}}
          />
        </Button>
      </span>
    </form>
  </div>
  <JSXStyle
    css=".KiwiLogin.jsx-30246891{width:480px;padding-top:128px;padding-left:40px;padding-right:40px;height:100%;overflow-y:auto;overflow-x:hidden;margin:auto;}p.title.jsx-30246891{color:#171b1e;font-size:28px;font-weight:bold;margin-bottom:8px;line-height:1.2;}.forgotPassword.jsx-30246891{margin-top:12px;display:inline-block;font-size:14px;color:#00a991;}.signIn.jsx-30246891{float:right;margin-bottom:20px;}form.jsx-30246891{margin-top:33px;}div.input.jsx-30246891{width:400px;height:44px;margin-bottom:34px;margin-top:4px;}label.jsx-30246891{width:352px;height:20px;font-size:14px;line-height:1.43;text-align:left;color:#46515e;display:inline-table;}div.errorMessage.jsx-30246891{margin-bottom:15px;}@media only screen and (max-width:480px){.KiwiLogin.jsx-30246891{width:100%;padding-top:64px;padding-left:0px;padding-right:0px;}p.title.jsx-30246891{font-size:22px;text-align:left;margin-top:0;}div.picture.jsx-30246891{display:none;}div.input.jsx-30246891{width:100%;margin-bottom:26px;}div.main.jsx-30246891{margin-left:16px;margin-right:16px;}div.forgotPassword.jsx-30246891{text-align:center;display:block;margin-bottom:20px;margin-top:0px;}.signIn.jsx-30246891{float:none;display:block;margin:0;}label.jsx-30246891{width:100%;}}@media only screen and (max-height:480px){.KiwiLogin.jsx-30246891{width:100%;padding-top:20px;}p.title.jsx-30246891{font-size:22px;text-align:center;margin-top:0;}div.picture.jsx-30246891{display:none;}div.main.jsx-30246891{width:288px;margin-left:auto;margin-right:auto;}div.input.jsx-30246891{width:288px;margin-bottom:26px;}div.forgotPassword.jsx-30246891{display:block;text-align:center;margin-bottom:20px;margin-top:0;}.signIn.jsx-30246891{float:none;display:block;margin:0;}}"
    styleId="30246891"
  />
</div>
`;
